crates/shared/src/event_handling.rs

```diff
@@ -108,7 +108,7 @@ where
             .past_events(&range)
             .await
             .context("failed to get past events")?
-            .ready_chunks(INSERT_EVENT_BATCH_SIZE)
+            .chunks(INSERT_EVENT_BATCH_SIZE)
```

Comment:
is this change because ready_chunks may yield an empty vector before the stream is over?

Reply:
Oh, forgot to explain this. ready_chunk is different because it will yield chunks when the underlying stream would be pending even if the chunk is not full yet. Neither will yield empty chunks and if they would it wouldn't be an issue. I just found the ready_chunk behavior a little surprising so I moved to chunks as it is closer to what this function on normal iterator does and I think this is what we intended.